lib/src/key/bf.rs

```diff
@@ -57,14 +57,16 @@ mod tests {
 	fn key() {
 		#[rustfmt::skip]
 		let val = Bf::new(
-			"test",
-			"test",
-			"test",
-			"test",
-			1,
-			2
+			"testns",
+			"testdb",
+			"testtb",
+			"testix",
+			7,
+			0x0d, // 13
 		);
 		let enc = Bf::encode(&val).unwrap();
+		assert_eq!(enc, b"/*testns*testdb*testtb!bftestix*\x07\x0d");
```

Comment:
From running `cargo test -p surrealdb -F kv-rocksdb,kv-mem key::bf`

```
---- key::bf::tests::key stdout ----
thread 'key::bf::tests::key' panicked at 'assertion failed: `(left == right)`
  left: `[47, 42, 116, 101, 115, 116, 110, 115, 0, 42, 116, 101, 115, 116, 100, 98, 0, 42, 116, 101, 115, 116, 116, 98, 0, 33, 98, 102, 116, 101, 115, 116, 105, 120, 0, 42, 0, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 13]`,
 right: `[47, 42, 116, 101, 115, 116, 110, 115, 42, 116, 101, 115, 116, 100, 98, 42, 116, 101, 115, 116, 116, 98, 33, 98, 102, 116, 101, 115, 116, 105, 120, 42, 7, 13]`', lib/src/key/bf.rs:68:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
@phughk the test on line 68 is incorrect. `0` bytes are added to terminate strings - this is necessary, because in order to guarantee lexicographic sort ordering of strings, we can't prefix a string in a key with a length. So we have to prevent `0` bytes inside strings, and wait until we encounter a `0` in order to 'terminate' the string. With regards to this example, `term_id`, and `doc_id` are `u64`, so they will always be encoded as 64bit (8 bytes), and therefore are prefixed with `0` bytes. Testing for this, is best left to the `storekey` crate.